tests/e2e/specs/upgrade/upload-plugin-new-version.test.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+import path from 'path';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	visitAdminPage,
+	installPlugin,
+	uninstallPlugin,
+	deactivatePlugin,
+	__experimentalRest as rest,
+} from "@wordpress/e2e-test-utils";
+import { addQueryArgs } from "@wordpress/url";
+
+async function getPlugins() {
+	const response = await rest({
+		method: 'GET',
+		path: "/wp/v2/plugins",
+	});
+	return response;
+}
+
+async function checkPluginStatus(pluginName) {
+	const plugins = await getPlugins();
+	const plugin = plugins.find(plugin => plugin.name === pluginName);
+
+	return plugin ? plugin.status : 'uninstalled';
+}
+
+async function checkAndInstallPlugin(pluginSlug, pluginName) {
+	const pluginStatus = await checkPluginStatus(pluginName);
+
+	if (pluginStatus === 'active') {
+		await deactivatePlugin(pluginSlug);
+		await uninstallPlugin(pluginSlug);
+		await installPlugin(pluginSlug, pluginName);
+	} else if (pluginStatus === 'inactive') {
+		await uninstallPlugin(pluginSlug);
+		await installPlugin(pluginSlug, pluginName);
+	} else {
+		await installPlugin(pluginSlug, pluginName);
+	}
```

Comment:
We could just do this, might be simpler:

```suggestion
	if (pluginStatus === 'active') {
		await deactivatePlugin(pluginSlug);
	}
	if (pluginStatus === 'inactive') {
		await uninstallPlugin(pluginSlug);
	}
	await installPlugin(pluginSlug, pluginName);
```